src/main/java/org/apache/mesos/hdfs/Scheduler.java

```diff
@@ -452,7 +454,7 @@ private boolean tryToLaunchNameNode(SchedulerDriver driver, Offer offer) {
     List<String> deadNameNodes = persistentState.getDeadNameNodes();
 
     if (deadNameNodes.isEmpty()) {
-      if (liveState.getNameNodeSize() == HDFSConstants.TOTAL_NAME_NODES) {
+      if (persistentState.getNameNodes().size() == HDFSConstants.TOTAL_NAME_NODES) {
```

Comment:
but for efficiency, the livestate is effectively the same since we reconcile tasks.